hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/CockroachLegacyDialect.java

```diff
@@ -0,0 +1,805 @@
+import static org.hibernate.type.SqlTypes.*;
+import static org.hibernate.type.descriptor.DateTimeUtils.appendAsDate;
+import static org.hibernate.type.descriptor.DateTimeUtils.appendAsTime;
+import static org.hibernate.type.descriptor.DateTimeUtils.appendAsTimestampWithMicros;
+
+/**
+ * A {@linkplain Dialect SQL dialect} for CockroachDB.
+ *
+ * @author Gavin King
+ */
+public class CockroachLegacyDialect extends Dialect {
+
+	private static final CockroachDBIdentityColumnSupport IDENTITY_COLUMN_SUPPORT = new CockroachDBIdentityColumnSupport();
+	// KNOWN LIMITATIONS:
+	// * no support for java.sql.Clob
+
+	private final PostgreSQLDriverKind driverKind;
+
+	public CockroachLegacyDialect() {
+		this( DatabaseVersion.make( 19, 2 ) );
+	}
+
+	public CockroachLegacyDialect(DialectResolutionInfo info) {
+		super(info);
+		driverKind = PostgreSQLDriverKind.determineKind( info );
+	}
+
+	public CockroachLegacyDialect(DatabaseVersion version) {
+		super(version);
+		driverKind = PostgreSQLDriverKind.PG_JDBC;
+	}
+
+	public CockroachLegacyDialect(DatabaseVersion version, PostgreSQLDriverKind driverKind) {
+		super(version);
+		this.driverKind = driverKind;
+	}
+
+	@Override
+	protected String columnType(int sqlTypeCode) {
+		switch ( sqlTypeCode ) {
+			case TINYINT:
+				return "smallint"; //no tinyint
+
+			case CHAR:
+			case NCHAR:
+			case VARCHAR:
+			case NVARCHAR:
+				return "string($l)";
+
+			case NCLOB:
+			case CLOB:
+				return "string";
+
+			case BINARY:
+			case VARBINARY:
+			case BLOB:
+				return "bytes";
+
+			case TIMESTAMP_UTC:
+				return columnType( TIMESTAMP_WITH_TIMEZONE );
+		}
+		return super.columnType( sqlTypeCode );
+	}
+
+	@Override
+	protected String castType(int sqlTypeCode) {
+		switch ( sqlTypeCode ) {
+			case CHAR:
+			case NCHAR:
+			case VARCHAR:
+			case NVARCHAR:
+			case LONG32VARCHAR:
+			case LONG32NVARCHAR:
+				return "string";
+			case BINARY:
+			case VARBINARY:
+			case LONG32VARBINARY:
+				return "bytes";
+		}
+		return super.castType( sqlTypeCode );
+	}
+
+	@Override
+	protected void registerColumnTypes(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
+		super.registerColumnTypes( typeContributions, serviceRegistry );
+		final DdlTypeRegistry ddlTypeRegistry = typeContributions.getTypeConfiguration().getDdlTypeRegistry();
+
+		ddlTypeRegistry.addDescriptor( new DdlTypeImpl( UUID, "uuid", this ) );
+		ddlTypeRegistry.addDescriptor( new DdlTypeImpl( GEOMETRY, "geometry", this ) );
+		ddlTypeRegistry.addDescriptor( new DdlTypeImpl( GEOGRAPHY, "geography", this ) );
+		ddlTypeRegistry.addDescriptor( new Scale6IntervalSecondDdlType( this ) );
+
+		// Prefer jsonb if possible
+		if ( getVersion().isSameOrAfter( 20 ) ) {
+			ddlTypeRegistry.addDescriptor( new DdlTypeImpl( INET, "inet", this ) );
+			ddlTypeRegistry.addDescriptor( new DdlTypeImpl( JSON, "jsonb", this ) );
+		}
+		else {
+			ddlTypeRegistry.addDescriptor( new DdlTypeImpl( JSON, "json", this ) );
```

Comment:
Looks like something got lost after the rebase:

```suggestion
		if ( PostgreSQLPGObjectJdbcType.isUsable() ) {
			// The following DDL types require that the PGobject class is usable/visible
			ddlTypeRegistry.addDescriptor( new DdlTypeImpl( GEOMETRY, "geometry", this ) );
			ddlTypeRegistry.addDescriptor( new DdlTypeImpl( GEOGRAPHY, "geography", this ) );
			ddlTypeRegistry.addDescriptor( new Scale6IntervalSecondDdlType( this ) );

			// Prefer jsonb if possible
			if ( getVersion().isSameOrAfter( 20 ) ) {
				ddlTypeRegistry.addDescriptor( new DdlTypeImpl( INET, "inet", this ) );
				ddlTypeRegistry.addDescriptor( new DdlTypeImpl( JSON, "jsonb", this ) );
			}
			else {
				ddlTypeRegistry.addDescriptor( new DdlTypeImpl( JSON, "json", this ) );
			}
```